### PR TITLE
Invoke VMSS parallel update API when vmss tag exists

### PR DESCRIPTION
**Pull request by lodrem**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
It reads tag `aks-managed-coordination` from VMSS, and then only invoke parallel network profile update APIs when the tag exists (tag value could be omitted)
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_vmss.go [hunk 1]
```diff
@@ -1359,10 +1359,24 @@ func (ss *ScaleSet) EnsureHostsInPool(service *v1.Service, nodes []*v1.Node, bac
...
+			}
+
+			batchSize, err := ss.VMSSBatchSize(meta.vmssName)
+			if err != nil {
+				klog.ErrorS(err, "Failed to get vmss batch size", logFields...)
+				return err
+			}
+
+			klog.V(2).InfoS("Begin to update VMs for VMSS with new backendPoolID", logFields...)
+			rerr := ss.VirtualMachineScaleSetVMsClient.UpdateVMs(ctx, meta.resourceGroup, meta.vmssName, update, "network_update", batchSize)
```
Comment (nilo19) [inline pkg/provider/azure_vmss.go:1377]: I think the code will look simpler if we check the tag in the vmss VM client.

**Comment by nilo19** [comment]
Please don't forget to squash the commits.

File: pkg/provider/azure_vmss.go (hunk 1 above)
Comment (lodrem) [reply pkg/provider/azure_vmss.go:1377]: I assume we need to get VMSS through cache. Could we access the cache in the vmss VM client?

File: pkg/provider/azure_vmss.go [hunk 2]
```diff
@@ -1825,3 +1853,17 @@ func (ss *ScaleSet) GetNodeVMSetName(node *v1.Node) (string, error) {
 	klog.V(4).Infof("ss.GetNodeVMSetName: found vmss name %s from node name %s", vmssName, node.Name)
 	return vmssName, nil
 }
+
+// VMSSBatchSize returns the batch size for VMSS operations.
+func (ss *ScaleSet) VMSSBatchSize(vmssName string) (int, error) {
+	batchSize := 1
```
Comment (feiskyer) [inline pkg/provider/azure_vmss.go:1859]: should we use the same default value `0` here as the cloud config file?

File: pkg/provider/azure_vmss.go (hunk 1 above)
Comment (feiskyer) [reply pkg/provider/azure_vmss.go:1377]: nit: the tag doesn't exist on VMs, it's only on VMSS

File: pkg/provider/azure_vmss.go (hunk 2 above)
Comment (lodrem) [reply pkg/provider/azure_vmss.go]: sure and fixed

**Comment by feiskyer** [comment]
@lodrem @MartinForReal release-1.1 failed with conflicts

**Comment by lodrem** [comment]
> @lodrem @MartinForReal release-1.1 failed with conflicts

Will cherry-pick for it manually. Thanks for reminding me.

**Pull request merged**